core/src/test/java/org/apache/iceberg/TestTables.java

```diff
@@ -93,6 +93,26 @@ public static TestTable create(
     return new TestTable(ops, name, reporter);
   }
 
+  public static TestTable create(
+      File temp,
+      String name,
+      Schema schema,
+      PartitionSpec spec,
+      int formatVersion,
+      Map<String, String> properties) {
+    TestTableOperations ops = new TestTableOperations(name, temp);
+    if (ops.current() != null) {
+      throw new AlreadyExistsException("Table %s already exists at location: %s", name, temp);
```

Comment:
Why this check? I think it makes sense but wondering why it's being added now

Reply:
Just followed other create methods. Example line 85